relayer/src/parachain.ts

```diff
@@ -50,14 +54,14 @@ class Parachain {
                 }
                 return response.json();
             })
-            .then(({ result }) => {
-                if (!result) {
-                    throw new Error(`Could not fetch ${this.chain} parablock ${hash} from ${this.url}`);
+            .then((data) => {
+                if (!data.result) {
+                    throw new Error(`Could not fetch ${this.chain} parablock ${hash}. Response: ${JSON.stringify(data)}`);
                 }
-                return result;
+                return data.result;
```

Comment:
This is better, but still lacks response validation

Reply:
do you mean checking if `response.result` is a valid block?

Reply:
Yes. At least verify that it has the data structure you expect because above check with succeed if it happens to return `{result: true}`, which is obviously not something you want.

Reply:
agree, thanks